### PR TITLE
Fixed hex parsing

### PR DESCRIPTION
**Pull request by davidepatti**
This PR solves the following:

- In .data section, hex strings unreasonable long hex strings are checked before any conversion. For example, there's no chance of trying to apply conversion to something that is > 18 chars trimmed (8 bytes + 0x).
- In .data section, each hex string is checked against its actual size  (byte, word16 etc...) , not according some converted value. You should be able to write .byte 0xff to put 8 ones somewhere.
- In code, now is possible to use all the 16 bits of immediate in hex strings
- The mem section in webUI suffered a little bug in displaying big hex values, due to unmotivated usage of a different method rather the re-using the same value used in the Java version.

### CODE REVIEW

**Comment by davidepatti** [comment]
Added test for these special cases handling. All the other pre-existing tests working as usual, which confirms that these edge cases were just not handled and tested.

File: src/test/java/org/edumips64/core/ParserTest.java [hunk 1]
```diff
@@ -271,4 +271,23 @@ public void MissingParmeterTest() throws Exception {
...
     parseCode("LW r1");
   }
+
+  @Test(expected = ParserMultiException.class)
+  public void unlimitedHexStringTest() throws Exception {
+    parseData(".word 0xdeadbeeffffffffffffffffff");
+  }
+
+  @Test
+  public void hexStringLimitData() throws Exception {
```
Comment (lupino3) [inline src/test/java/org/edumips64/core/ParserTest.java:281]: Note that this test will "succeed" if any of the below `parseData` calls raises an exception. If you expect all of them to raise exceptions, they need to be separate tests.

File: src/test/java/org/edumips64/core/ParserTest.java [hunk 2]
```diff
@@ -271,4 +271,23 @@ public void MissingParmeterTest() throws Exception {
     // Fully missing parameter. Before the fix for #1048, this would throw StringIndexOutOfBoundsException.
     parseCode("LW r1");
   }
+
```
Comment (lupino3) [inline src/test/java/org/edumips64/core/ParserTest.java:274]: Please add a comment explaining that the below are regression tests for #1317

File: src/main/java/org/edumips64/client/ResultFactory.java [hunk 3]
```diff
@@ -109,15 +110,19 @@ private String getMemory() {
 
             JSONArray cellArray = new JSONArray();
             cells.forEach((address, element) -> {
-                cellArray.set(cellArray.size(), new FluentJsonObject()
-                        .put("address_hex", element.getAddressHex())
-                        .put("address", element.getAddress())
-                        .put("value", element.getValue())
-                        .put("value_hex", element.getValueHex())
```
Comment (lupino3) [inline src/main/java/org/edumips64/client/ResultFactory.java:116]: Do we still use/need `getValueHex()`? If not, can you please delete it?

**Review by lupino3 — CHANGES_REQUESTED** [review]
Looks good! Just a couple of minor comments.

File: src/test/java/org/edumips64/core/ParserTest.java (hunk 1 above)
Comment (davidepatti) [reply src/test/java/org/edumips64/core/ParserTest.java:281]: This test does not expect any execption. To succeed, alla the parseData must be ok.

Comment (lupino3) [reply src/test/java/org/edumips64/core/ParserTest.java:282]: Right, sorry!

**Review by lupino3 — CHANGES_REQUESTED** [review]
@davidepatti there are 2 edge cases not covered by unit tests (see Codecov patch comments on `Parser.java`). If you have time, could you add 2 unit tests to cover them? I feel they are important enough that they need to be represented in `ParserTest.java`. Other than that, the PR is good to go. Thanks!

**Comment by lupino3** [comment]
Thanks!
